wagtail/snippets/models.py

```diff
@@ -140,3 +143,25 @@ def register_deferred_snippets():
     DEFER_REGISTRATION = False
     for model, viewset in DEFERRED_REGISTRATIONS:
         _register_snippet_immediately(model, viewset)
+
+
+def create_extra_permissions(*args, **kwargs):
+    content_types = ContentType.objects.get_for_models(
+        *SNIPPET_MODELS, for_concrete_models=False
+    ).values()
+    model_cts = dict(zip(SNIPPET_MODELS, content_types))
+
+    permissions = []
+    for model, ct in model_cts.items():
+        if issubclass(model, DraftStateMixin):
```

Comment:
When we add support for `LockableMixin`, we can add `lock` and `unlock` permissions with another `if` check in this same function